internal/oauth/oauth.go

```diff
@@ -58,10 +64,16 @@ const (
 	GovClientID = "0oabtyfelbTBdoucy297" // GovClientID for production
 )
 
-func changeHostNameToContainer() {
-	if !strings.Contains(config.UserAgent, config.ContainerHostName) {
-		config.UserAgent = strings.ReplaceAll(config.UserAgent, config.HostName, config.ContainerHostName)
-		config.HostName = config.ContainerHostName
+func patchConfigHostname() {
+	hostnameEnvVal := os.Getenv(config.HostnameEnv)
+
+	if hostnameEnvVal == "" && (os.Getenv("MONGODB_ATLAS_IS_CONTAINERIZED") == "true") {
```

Comment:
Does it make sense to consider the `MONGODB_ATLAS_IS_CONTAINERIZED` as true when is defined instead of explicitly check for its value? I am wondering if there is a scenario where the var is defined with a different value than true.